fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1346,6 +1353,87 @@ def assert_state(self, state: Union[TrainingState, List[TrainingState]]) -> None
                 traceback.print_stack()
             raise ValueError(msg)
 
+    def _consolidate_optim_state_dict(
+        self, optim: torch.optim.Optimizer, recipient_rank: Optional[int] = None
+    ) -> List[Dict]:
+        """Update the consolidated state_dict list, one per rank.
```

Comment:
should this be called only on the root FSDP instance?

Reply:
Yes, more specifically it should be called on the instance that was the argument to `optimizer(model.parameters()`. Are there other cases?